### DISCUSSION TITLE
Grafana is not querying with a delta timestamp while auto-refreshing

### DISCUSSION THREAD

**Discussion by CostinSM** [grafana/grafana-csv-datasource#102] [Q&A]
Hello Marcus,

I've been using your CSV Data Source and it has been working wonderfully. However, I have encountered an issue with the call Grafana is making to our API. I noticed that if I select any time interval for the data to be automatically refreshed, the calls received by the API from Grafana do not have a delta time interval, meaning that with every call, all the data is retrieved instead of just the newly updated one.
I have to mention that I am using "from" and "to" parameters in the params tab of the Data Source, which are global parameters that Grafana uses and not something custom to my case.
Do you know if this is an issue Grafana has? Or is it related to the Data Source?

Also, I attended the seminar you recently hosted for getting started with Grafana.. Nice job! Plenty of useful information in it!

Regards,
Costin

**Comment by marcusolsson** [comment]
Unfortunately, the plugin doesn't support filtering of any kind at this point. There's an [open feature request](https://github.com/marcusolsson/grafana-csv-datasource/issues/58) to add support for a query language such as `csvq`, that would let you specify the time interval to use.

We could solve this without introducing csvq though. But since a query can return multiple time fields, we'd have to introduce an option specify which field you'd like to use for filtering (if there's only one, we could default to that).

Even so, the backend would still have to get the entire CSV file to be able to do any filtering, but we could limit the data returned to the browser at least.

Thanks for the kind words! I'm glad the seminar was helpful to you! ❤️

**Reply by CostinSM** [reply]
Indeed I have only one time field that could simplify things a little bit.
The thing is I am generating the CSV on the fly, new data is getting added to it constantly and when the call from Grafana comes, the global variables I have do manage to filter the data for the specified time interval. The problem is when the automatic refresh interval is set (let's say 5 min), the calls are made (probably using the same global varables) for the whole range that was previously available + 5min and so on, not only the last 5 min.

![Grafana params](https://user-images.githubusercontent.com/87074202/124768045-0124f680-df41-11eb-8926-84142b7c8185.png)

Also, now that we are on the subject :D.. Right now I am using one panel (the one in thepicture) to query all the fields from the source CSV and all the other panels are using this initial panel as a data source, filtering only the needed fields. Is there a better way to do this, maybe something like the annotations query, that is built into the dashboard? This would be of great help as well because I would like to limit the number of calls to the API and due to the fact that I have to use Transform in all the oher panels, I cannot set any Alerts to them..

Regards,
Costin
